pennylane/qinfo/transforms.py

```diff
@@ -731,17 +734,33 @@ def circ(params):
 
     """
 
-    if qnode.device.shots and isinstance(qnode.device, (DefaultQubitLegacy, DefaultQubit)):
+    if device.shots and isinstance(device, (DefaultQubitLegacy, DefaultQubit)):
+        tapes, processing_fn = metric_tensor(tape, *args, **kwargs)
 
-        def wrapper(*args0, **kwargs0):
-            return 4 * metric_tensor(qnode, *args, **kwargs)(*args0, **kwargs0)
+        def processing_fn_multiply(res):
+            res = qml.execute(res, device=device)
+            return 4 * processing_fn(res)
 
-    else:
+        return tapes, processing_fn_multiply
 
-        def wrapper(*args0, **kwargs0):
-            return 4 * adjoint_metric_tensor(qnode, *args, **kwargs)(*args0, **kwargs0)
+    res = adjoint_metric_tensor(tape, *args, **kwargs)
```

Comment:
Shouldn't this also be `tapes, processing_fn` instead of res?

Reply:
Nevermind I forgot that we return the processed results for informative transforms.